bluemira/base/reactor.py

```diff
@@ -139,17 +153,76 @@ def _filter_tree(self, comp: Component, dims_to_show: Tuple[str, ...]) -> Compon
         as filtering would mutate the ComponentMangers' underlying component trees
         """
         comp_copy = comp.copy()
-        comp_copy.filter_components(dims_to_show)
+        comp_copy.filter_components(dims_to_show, filter_)
         return comp_copy
 
-    def _plot_dims(self, comp: Component, dims_to_show: Tuple[str, ...]):
+    def _plot_dims(
+        self,
+        comp: Component,
+        dims_to_show: Tuple[str, ...],
+        filter_: Union[Callable[[PhysicalComponent], bool], None],
+    ):
         for i, dim in enumerate(dims_to_show):
             ComponentPlotter(view=dim).plot_2d(
-                self._filter_tree(comp, dims_to_show),
+                self._filter_tree(comp, dims_to_show, filter_),
                 show=i == len(dims_to_show) - 1,
             )
 
 
+class FilterMaterial:
+    """
+    Filter nodes by material
+
+    Parameters
+    ----------
+    material:
+       materials to include
+    not_material:
+       materials to exclude
+
+    """
+
+    def __init__(
+        self,
+        material: Union[
+            Type[SerialisedMaterial], Tuple[Type[SerialisedMaterial]], None
+        ] = None,
+        not_material: Union[
+            Type[SerialisedMaterial], Tuple[Type[SerialisedMaterial]], None
+        ] = Void,
+    ):
+        super().__setattr__("material", material)
+        super().__setattr__("not_material", not_material)
+
+    def __call__(self, node: anytree.Node) -> bool:
+        """Filter node based on material include and exclude rules"""
+        return hasattr(node, "material") and self._filterer(node.material)
+
+    def __setattr__(self, name: str, value: Any):
+        """
+        Override setattr to force immutability
+
+        This method makes the class nearly immutable as no new attributes
+        can be modified or added by standard methods.
+
+        See #2236 discussion_r1191246003 for further details
+        """
+        raise AttributeError(f"{type(self).__name__} is immutable")
+
+    def _filterer(
```

Comment:
Think the name is bit funny, maybe _apply_filters?

Reply:
I suppose, less fun though :smile: